vulkan_helpers/vulkan_application.h

```diff
@@ -260,8 +260,10 @@ class PipelineLayout {
       raw_layouts.push_back(descriptor_set_layouts_.back());
     }
 
-    containers::vector<VkPushConstantRange> push_constant_ranges(ranges.begin(),
-                                                            ranges.end());
+    containers::vector<VkPushConstantRange> push_constant_ranges(allocator);
```

Comment:
Isn't there a constructor that takes a begin, end and allocator?

Reply:
There should be 
```
template< class InputIt >
vector( InputIt first, InputIt last,
        const Allocator& alloc = Allocator() );
```

Reply:
Fixed